### DISCUSSION TITLE
RPM config(noreplace) behavior in deb

### DISCUSSION THREAD

**Discussion by oliveromahony** [goreleaser/nfpm#591] [Q&A]
Hi there,

I have a configuration bundled in my packages similar to that specified in https://github.com/goreleaser/nfpm/blob/main/www/docs/configuration.md

The relevant configuration is as follows:
```
- src: path/to/deb/file.conf
    dst: /etc/file.conf
    type: config|noreplace
    packager: deb
```

When I generate a deb package, I still get prompted on an install 
 ```
apt-get install -f -y ./binary_amd64.deb
 ```

 ```
Configuration file '/etc/file.conf'
 ==> File on system created by you or by a script.
 ==> File also in package provided by package maintainer.
   What would you like to do about it ?  Your options are:
    Y or I  : install the package maintainer's version
    N or O  : keep your currently-installed version
      D     : show the differences between the versions
      Z     : start a shell to examine the situation
 The default action is to keep your current version.
*** file.conf (Y/I/N/O/D/Z) [default=N] ?
 ```

Is it possible to configure nfpm to have the same behavior as %config(noreplace) in RPM like this https://www.cl.cam.ac.uk/~jw35/docs/rpm_config.html ? Where the old configuration is not modified and the user is not prompted on a package install in debian based packages?

NFPM: v2.18.0
go: 1.19.3

**Comment by caarlos0** [comment]
nfpm maps `config|noreplace` to `%config(noreplace)` in rpm, so it should work, you just need to remove the `packager: deb`, probably

**Reply by oliveromahony** [reply]
Thanks for the reply, I originally had the following config:

```
- src: path/to/deb/file.conf
    dst: /etc/file.conf
    type: config|noreplace
```
and the behavior was the same unfortunately

**Reply by djgilcrease** [reply]
debian does not support anything like `config|noreplace` at build time. They do support not prompting, but only by specifying command line options at install time.

https://raphaelhertzog.com/2010/09/21/debian-conffile-configuration-file-managed-by-dpkg/

**Reply by oliveromahony** [reply]
yeah that's why I created a discussion and not a bug. I know I could do it the way mentioned, however I wanted NFPM to do it for me if possible
